### PR TITLE
MGMT-16841: Failed to add host when infraenv has networkType of DHCP

### PR DESCRIPTION
**Pull request by jgyselov**
https://issues.redhat.com/browse/MGMT-16841

### CODE REVIEW

**Review by ammont82 — DISMISSED** [review]

File: libs/ui-lib/lib/cim/components/Agent/BMCForm.tsx [hunk 1]
```diff
@@ -187,14 +187,14 @@ const emptyValues: AddBmcValues = {
...
     'interfaces:\n\
-- name: <nic1_name>\n\
-  type: ethernet\n\
-  state: up\n\
-  ipv4:\n\
-    address:\n\
-    - ip: <ip_address>\n\
-      prefix-length: 24\n\
-    enabled: true\n\
+  - name: <nic1_name>\n\
```
Comment (mareklibra) [inline libs/ui-lib/lib/cim/components/Agent/BMCForm.tsx:190]: It's safer to use multi-line template literal to preserve formatting. We can avoid explicit `\n` and be sure about white spaces at the beginning of a line.

File: libs/ui-lib/lib/cim/components/Agent/BMCForm.tsx [hunk 2]
```diff
@@ -228,8 +231,13 @@ const getInitValues = (
       macMapping: nmState?.spec?.interfaces || [{ macAddress: '', name: '' }],
     };
   } else {
-    return emptyValues;
+    values = emptyValues;
+  }
+
+  if (!addNMState) {
+    values.nmState = '';
```
Comment (mareklibra) [inline libs/ui-lib/lib/cim/components/Agent/BMCForm.tsx:238]: It's probably ok for recent flow, but the `addNMState` and `isEdit` are theoretically independent on each other and so the `values` can be `undefined` here.

What about changing the line above to:
```
let values = emptyValues;
```

**Review by ammont82 — APPROVED** [review]

File: libs/ui-lib/lib/cim/components/Agent/BMCForm.tsx [hunk 3]
```diff
@@ -180,26 +180,25 @@ const emptyValues: AddBmcValues = {
...
-  config:\n\
-    server:\n\
-    - <dns_ip_address>\n\
-routes:\n\
-  config:\n\
-  - destination: 0.0.0.0/0\n\
-    next-hop-address: <next_hop_ip_address>\n\
-    next-hop-interface: <next_hop_nic1_name>\n\
-',
+  nmState: `interfaces:
```
Comment (mareklibra) [inline libs/ui-lib/lib/cim/components/Agent/BMCForm.tsx:183]: nit: if the `interfaces:` starts on a new line, the formatting will be easier to read. But LGTM.

**Review by mareklibra — APPROVED** [review]

**Pull request merged**